content/en/blog/_posts/2024-11-11-dynamic-kubernetes-api-server-for-cozystack/index.md

```diff
@@ -0,0 +1,454 @@
+---
+layout: blog
+title: "How we built a dynamic Kubernetes API Server for the API Aggregation Layer in Cozystack"
+slug: dynamic-kubernetes-api-server-for-cozystack
+date: 2024-11-11
+author: >
+  Andrei Kvapil (Ænix)
+---
+
+Hi there! I'm Andrey Kvapil, but you might know me as [@kvaps](https://github.com/kvaps) in communities dedicated to Kubernetes
+and cloud-native tools. In this article, I want to share how we implemented our own extension api-server
+in the open-source PaaS platform, Cozystack.
+
+Kubernetes truly amazes me with its powerful extensibility features. You're probably already
+familiar with the [operator pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/)
+and frameworks like [kubebuilder](https://book.kubebuilder.io/) and
+[operator-sdk](https://sdk.operatorframework.io/) that help you implement it. In a nutshell, they
```

Comment:
You're really talking about controllers, so:
```suggestion
familiar with the [controller](/docs/concepts/architecture/controller/) concept
and frameworks like [kubebuilder](https://book.kubebuilder.io/) and
[operator-sdk](https://sdk.operatorframework.io/) that help you implement it. In a nutshell, they
```

Yes, the `operator-sdk` lets you write operators, but only because all operators are also controllers.